lib/proxy/request.rb

```
@@ -1,23 +1,57 @@
 require 'net/http'
 require 'net/https'
+require 'addressable/uri'
 require 'uri'
 
 # TODO: need settings validation on startup, otherwise we get a 500 error due to missing/wrong config settings when api is accessed
 
 module Proxy::HttpRequest
-  class ForemanRequest
-    def send_request(path, body)
-      path = [uri.path, path].join('/') unless uri.path.empty?
-      req = Net::HTTP::Post.new(URI.join(uri.to_s, path).path)
+  class ForemanRequestFactory
+    def initialize(base_uri)
+      @base_uri = base_uri
+    end
+
+    def create_get(path, query={})
+      uri = uri(path)
+      uri.query_values = query
+      req = Net::HTTP::Get.new(uri.path)
```

if the uri is 'https://google.com/'
the request it would make to the server would be
GET https://google.com/ 
instead of 
GET /

updating since we're moving back to URI instead of Addressable::URI

i saw that it was present in settings.yml in develop branch...so thought it would be  to use it.
going to use a fail fast( couldn't find an existing example yet).